Micronaut 3.8.5 fails to start when using programmatic logger configuration on logback > 1.2.11

### Expected Behavior

Micronaut application should start when using a recent logback version and log levels defined within the application.xml file, or give an appropriate error message.

### Actual Behaviour

Silent application startup failure first observed on 3.8.4 (similar to #8758), is still present on 3.8.5. Issue only occurs when logger level(s) are defined within the application.xml, issue is not present when these are removed. Switching to logback 1.2.11 is tested to resolve the issue. 

The following output is seen when running the application, the error is silent:
```
DEBUG i.m.context.DefaultBeanContext - Resolving beans for type: io.micronaut.logging.LoggingSystem
DEBUG i.m.context.DefaultBeanContext - Finding candidate beans for type: LoggingSystem E
DEBUG i.m.context.condition.Condition - Bean [Definition: io.micronaut.management.endpoint.loggers.impl.LogbackLoggingSystem] will not be loaded due to failing conditions:
DEBUG i.m.context.condition.Condition - * No bean of type [class io.micronaut.management.endpoint.loggers.LoggersEndpoint] present within context
DEBUG i.m.context.condition.Condition - Bean [io.micronaut.management.endpoint.loggers.impl.$Log4jLoggingSystem$Definition] will not be loaded due to failing conditions:
DEBUG i.m.context.condition.Condition - * Class [org.apache.logging.log4j.core.LoggerContext] is not present
DEBUG i.m.context.condition.Condition - Bean [io.micronaut.logging.impl.$Log4jLoggingSystem$Definition] will not be loaded due to failing conditions:
DEBUG i.m.context.condition.Condition - * Class [org.apache.logging.log4j.core.config.Configurator] is not present
DEBUG i.m.context.DefaultBeanContext -   class io.micronaut.logging.impl.LogbackLoggingSystem null Definition: io.micronaut.logging.impl.LogbackLoggingSystem
DEBUG i.m.context.DefaultBeanContext - Finalized bean definitions candidates: [Definition: io.micronaut.logging.impl.LogbackLoggingSystem]
DEBUG i.m.context.DefaultBeanContext -   class io.micronaut.logging.impl.LogbackLoggingSystem null Definition: io.micronaut.logging.impl.LogbackLoggingSystem
DEBUG io.micronaut.context.lifecycle - Created bean [io.micronaut.logging.impl.LogbackLoggingSystem@2f09e6b2] from definition [Definition: io.micronaut.logging.impl.LogbackLoggingSystem] with qualifier [null]
DEBUG i.m.context.DefaultBeanContext - Found a registration BeanRegistration: io.micronaut.logging.impl.LogbackLoggingSystem@2f09e6b2 for candidate: Definition: io.micronaut.logging.impl.LogbackLoggingSystem with qualifier: null

Process finished with exit code 1
```

Debugging produces the following error:
```
ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [io.micronaut.logging.PropertiesLoggingLevelsConfigurer] could not be loaded: Error instantiating bean of type  [io.micronaut.logging.PropertiesLoggingLevelsConfigurer]

Message: 'void ch.qos.logback.classic.spi.Configurator.configure(ch.qos.logback.classic.LoggerContext)'
Path Taken: new PropertiesLoggingLevelsConfigurer(Environment environment,List loggingSystems)
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.logging.PropertiesLoggingLevelsConfigurer] could not be loaded: Error instantiating bean of type  [io.micronaut.logging.PropertiesLoggingLevelsConfigurer]

Message: 'void ch.qos.logback.classic.spi.Configurator.configure(ch.qos.logback.classic.LoggerContext)'
Path Taken: new PropertiesLoggingLevelsConfigurer(Environment environment,List loggingSystems)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1921)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:249)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3326)
	at io.micronaut.context.DefaultBeanContext.finalizeConfiguration(DefaultBeanContext.java:3684)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:341)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:194)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:323)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:309)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.logging.PropertiesLoggingLevelsConfigurer]

Message: 'void ch.qos.logback.classic.spi.Configurator.configure(ch.qos.logback.classic.LoggerContext)'
Path Taken: new PropertiesLoggingLevelsConfigurer(Environment environment,List loggingSystems)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2367)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2746)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1915)
	... 9 common frames omitted
Caused by: java.lang.NoSuchMethodError: 'void ch.qos.logback.classic.spi.Configurator.configure(ch.qos.logback.classic.LoggerContext)'
	at io.micronaut.logging.impl.LogbackUtils.programmaticConfiguration(LogbackUtils.java:96)
	at io.micronaut.logging.impl.LogbackUtils.configure(LogbackUtils.java:74)
	at io.micronaut.logging.impl.LogbackUtils.configure(LogbackUtils.java:53)
	at io.micronaut.logging.impl.LogbackLoggingSystem.refresh(LogbackLoggingSystem.java:57)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at io.micronaut.logging.PropertiesLoggingLevelsConfigurer.initLogging(PropertiesLoggingLevelsConfigurer.java:83)
	at io.micronaut.logging.PropertiesLoggingLevelsConfigurer.<init>(PropertiesLoggingLevelsConfigurer.java:67)
	at io.micronaut.logging.$PropertiesLoggingLevelsConfigurer$Definition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2354)
	... 16 common frames omitted
```

The return type of `Configurator.configure` was changed from `void` to `ExecutionStatus` in logback 1.3.0, so it seems this is another incompatbility caused by compiling and running against different versions of logback

### Steps To Reproduce

1. Create a micronaut 3.8.4/5 application using logback 1.4.5
2. Add a programmatic log level configuration to the application.xml file, as shown in the user documentation [here](https://docs.micronaut.io/latest/guide/#logging). Eg:
```
logger:
  levels:
    io.micronaut.http: DEBUG
```
3. Observe failure to start application when launched

### Environment Information

All operating systems
JDK 17

### Example Application

_No response_

### Version

3.8.5

--- a/src/main/docs/guide/logging/logback.adoc
+++ b/src/main/docs/guide/logging/logback.adoc
@@ -2,6 +2,8 @@ To use the logback library, add the following dependency to your build.
 
 dependency:ch.qos.logback:logback-classic[gradleScope="implementation"]
 
+NOTE: Logback 1.3.x+ included a breaking binary change that may prevent it working with 3.8.x of the Micronaut framework. If you are using Logback 1.3.x+ and are experiencing issues, please downgrade to Logback 1.2.x.
+
 If it does not exist yet, place a link:https://logback.qos.ch/manual/configuration.html[logback.xml] file in the resources folder and modify the content for your needs. For example:
 
 .src/main/resources/logback.xml

